tests/scene/test_option_button.h

```diff
@@ -0,0 +1,142 @@
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef TEST_OPTION_BUTTON_H
+#define TEST_OPTION_BUTTON_H
+
+#include "scene/gui/option_button.h"
+
+#include "tests/test_macros.h"
+
+namespace TestOptionButton {
+
+TEST_CASE("[SceneTree][OptionButton] Initialization") {
+	OptionButton *test_opt = memnew(OptionButton);
+
+	SUBCASE("There should be no options right after initialization") {
+		CHECK_FALSE(test_opt->has_selectable_items());
+		CHECK(test_opt->get_item_count() == 0);
+	}
+
+	memdelete(test_opt);
+}
+
+TEST_CASE("[SceneTree][OptionButton] Single item") {
+	OptionButton *test_opt = memnew(OptionButton);
+
+	SUBCASE("There should a single item after after adding one") {
+		test_opt->add_item("single", 1013);
+
+		CHECK(test_opt->has_selectable_items());
+		CHECK(test_opt->get_item_count() == 1);
+		CHECK(test_opt->get_item_index(1013) == 0);
+		CHECK(test_opt->get_item_id(0) == 1013);
+
+		test_opt->remove_item(0);
+
+		CHECK_FALSE(test_opt->has_selectable_items());
+		CHECK(test_opt->get_item_count() == 0);
+	}
+
+	SUBCASE("There should a single item after after adding an icon") {
+		Ref<Texture2D> *test_icon = memnew(Ref<Texture2D>);
```

Comment:
```suggestion
		Ref<Texture2D> test_icon = memnew(Texture2D);
```
Don't think it makes sense to create a pointer to a ref counted like this